src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -2798,7 +2798,7 @@ namespace QgsWms
                                           QStringLiteral( "error message: %1. The XML string was: %2" ).arg( errorMsg, filter ) );
           }
           QDomElement filterElem = filterXml.firstChildElement();
-          QScopedPointer<QgsExpression> expression( QgsOgcUtils::expressionFromOgcFilter( filterElem ) );
+          QScopedPointer<QgsExpression> expression( QgsOgcUtils::expressionFromOgcFilter( filterElem, dynamic_cast<QgsVectorLayer *>( layer ) ) );
```

Comment:
Is there any reason not to use `filteredLayer`? It would avoid to cast a second time.